File: Scraped/2022/src/day01/day01.ts
import { Task } from "../task";

export class Day01 extends Task {
    partOne(): String {
        let elfs = this.getElfs();

        return `${elfs[0]}`;
    }
    partTwo(): String {
        let elfs = this.getElfs();

        const result = elfs[0] + elfs[1] + elfs[2];

        return `${result}`;
    }
    getElfs(): number[] {
        const input = this.readInput();
        let calorieList = input.split('\n');

        let elfs : number[] = [];
        
        let sumCalories = 0;
        calorieList.forEach(element => {
            if(Number.isNaN(parseInt(element))) {
                elfs.push(sumCalories);
                sumCalories = 0;
            } else {
                sumCalories += parseInt(element);
            }
        });

        elfs.sort((n1,n2) => n2 - n1);
        return elfs;
    }
}

File: 2022/day01/Day01.cs
namespace AdventOfCode.Day01
{
    class Day01 : DailyTask
    {
        public override string PartOne()
        {
            return GetElfCalories().OrderDescending().FirstOrDefault().ToString();
        }

        public override string PartTwo()
        {
            var elfCalories = GetElfCalories();
            return elfCalories.OrderDescending().Take(3).Sum().ToString();
        }

        private List<int> GetElfCalories() {
            var splittedInput = Input.Split("\n");
            List<int> elfs = [];
            var sumCalories = 0;

            foreach (var stringCalories in splittedInput)
            {
                if (int.TryParse(stringCalories, out int calories))
                {
                    sumCalories += calories;
                }
                else
                {
                    elfs.Add(sumCalories);
                    sumCalories = 0;
                }
            }
            elfs.Add(sumCalories);

            return elfs;
        }
    }
 }